### ISSUE TITLE
Feature request: Create or Update record item in single method

### ISSUE THREAD

**Issue by mbecker** [pocketbase/js-sdk#12]
Hi,

sorry again to create a new feature request. I want to create an item or create if its exists. At the moment I'm listing the item with a filter (like "key = 'value'): If the response includes the single item I'm updating it; if not I'm creating it.

It would be quite nice to have a method as follows:
```typescript
/**
* Creates a new item.
*/
create(sub: string, bodyParams?: {}, queryParams?: {
  'filter': FILTEXPRESSION WHICH RETURNS A SINGLE ITEM
}): Promise<M>;
```

**Comment by ganigeorgiev** [comment]
In the above example is not very clear that the `filter` query parameter will be used for fetching an existing record.
Furthermore, the provided filter may not be sufficient or may result in matching multiple records and the definition of the method will become too broad since we'll have to decide what its behavior should be in this type of scenario, which probably not everyone will agree (someone may say that we will update only the first record, others will want to update all matching records, 
 etc.).

We can add an `upsert` method that do first `getOne` and then `create` or `update`, depending on the result, but I'm against it as it is better to be explicit to avoid ambiguities.

**Comment by mbecker** [comment]
Originally I had thought of the http verb PUT. Since this http verb is not present in the API documentation or source code I thought of an alternative so that there is no major change. I think my suggestion without an existing PUT makes no sense and can be left to the user/client; as in my case.

What do you think about implementing the http verb PUT in pocketbase for records?

**Comment by ganigeorgiev** [comment]
The HTTP verb doesn't really matter, it just semantic.

The main problem is that the `filter` in your example could resolve to more than 1 result.

I don't think your proposal is OK because it introduces ambiguities, without saving too much time and code for the developer.

**Comment by mbecker** [comment]
Yes, the http verb PUT is just semantic. But the common definition is as follows:

> The HTTP PUT request method creates a new resource or replaces a representation of the target resource with the request payload

[MDN Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT)

That's what I more or less looking for.

At the moment I don't know if it's possible to create a DB collection field `ID` defined by the user / migration files or if you can create the `ID` by sending as a JSON key/value it in the http body. 

(No access to the dev machine until Sunday evening to test it; sorry)
